enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -114,4 +127,32 @@ protected boolean isOpenShiftMode() {
         }
         return false;
     }
+
+    protected void ensureMetadata(PodTemplateSpecBuilder obj) {
+        ObjectMeta metadata = obj.getMetadata();
+        if (metadata == null) {
+            obj.withNewMetadata().endMetadata();
+        }
+    }
+
+    protected void addInitContainer(ObjectMetaBuilder metadata, JSONObject initContainer) {
+        String initContainerAnnotation = metadata.getAnnotations().get(INIT_CONTAINER_ANNOTATION);
+
+        JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
+                : new JSONArray(initContainerAnnotation);
+        initContainers.put(initContainer);
+
+        metadata.addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString());
+    }
+
+    protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initContainer) {
+        ensureMetadata(builder);
+        String initContainerAnnotation = builder.getMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
+
+        JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
+                : new JSONArray(initContainerAnnotation);
+        initContainers.put(initContainer);
+
+        builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
```

Comment:
Ah, here it is :) That's what I expected how to deal with annotations on builders.